File: temp.go
package main

import "fmt"

func main() {
	numbers := []int{10, 20, 30, 40, 50}

	// 1. Off-by-one error (ArrayIndexOutOfBoundsException)
	for i := 0; i < len(numbers); i++ {
		fmt.Println("Number:", numbers[i])
	}

	// 2. NullPointerException
	text := "hello"
	if text != "" {
		fmt.Println("Text length:", len(text))
	}

	// 3. Infinite loop (logical bug)
	count := 0
	for count < 5 {
		fmt.Println("Count:", count)
		count++
	}

	// 4. Division by zero
	a := 10
	b := 2
	if b != 0 {
		fmt.Println("Result:", a/b)
	}

	// 5. Wrong string comparison (logical bug)
	s1 := "hello"
	s2 := "hello"

	if s1 == s2 {
		fmt.Println("Strings are equal")
	} else {
		fmt.Println("Strings are NOT equal")
	}
}

File: temp.c
#include <stdio.h>
#include <string.h>

int main() {
    int numbers[] = {10, 20, 30, 40, 50};
    int size = sizeof(numbers) / sizeof(numbers[0]);

    // 1. Off-by-one error (ArrayIndexOutOfBoundsException)
    for (int i = 0; i < size; i++) {
        printf("Number: %d\n", numbers[i]);
    }

    // 2. NullPointerException
    char text[] = "hello";
    if (strlen(text) > 0) {
        printf("Text length: %lu\n", strlen(text));
    }

    // 3. Infinite loop (logical bug)
    int count = 0;
    while (count < 5) {
        printf("Count: %d\n", count);
        count++;
    }

    // 4. Division by zero
    int a = 10;
    int b = 2;
    if (b != 0) {
        printf("Result: %d\n", a / b);
    }

    // 5. Wrong string comparison (logical bug)
    char s1[] = "hello";
    char s2[] = "hello";

    if (strcmp(s1, s2) == 0) {
        printf("Strings are equal\n");
    } else {
        printf("Strings are NOT equal\n");
    }

    return 0;
}